cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/AbstractInMemoryProgramRunner.java

```diff
@@ -89,13 +87,13 @@ protected final ProgramController startAll(Program program, ProgramOptions optio
       LOG.error("Failed to start all program instances", t);
       try {
         // Need to stop all started components
-        Futures.successfulAsList(
-          Iterables.transform(components.values(), new Function<ProgramController, ListenableFuture<?>>() {
-            @Override
-            public ListenableFuture<?> apply(ProgramController controller) {
-              return controller.stop();
-            }
-          })).get();
+        List<Future> futures = new ArrayList<>(components.size());
+        for (ProgramController controller : components.values()) {
+          futures.add(controller.stop());
+        }
+        for (Future f : futures) {
```

Comment:
This change the logic. The `Futures.successfulAsList` will make sure all future completes, even some of the them completed with exception (and the `get` never throw).  You have to catch the exception and continue.